xlml/utils/tpu.py

```diff
@@ -315,4 +321,8 @@ def ssh_tpu(
           )
       },
   )
-  ssh_group.run(cmds)
+
+  queued_resource_name = qualified_name.split('/')[-1]
+  # write stderr and output into a log file
+  run_cmds = f'set -euxo pipefail; ({cmds}) 2>&1 | tee -a /tmp/{queued_resource_name}'
```

Comment:
Is this shell `bash`? `|&` combine both stderr and stdout https://stackoverflow.com/a/35385038